pypeit/scripts/cache_github_data.py

```diff
@@ -1,99 +1,141 @@
+        valid_paths = list(github_paths.keys())
+        if any([path not in valid_paths for path in args.exclude]):
+            raise ValueError('Provided path to exclude not valid.  Options are: '
+                             f'{", ".join(valid_paths)}')
+        include_paths = ['all'] + valid_paths
+        if args.include is not None and any([path not in include_paths for path in args.include]):
+            raise ValueError('Provided path to include not valid.  Options are: '
+                             f'{", ".join(include_paths)}')
+        selected_paths = np.setdiff1d(valid_paths, args.exclude).tolist() \
+                            if args.include is None else args.include
+        if 'all' in selected_paths:
+            selected_paths = valid_paths
+        
+        # Get the relevant github branch
+        branch = cache.git_branch()
 
-            # Make the list of files to get
-            dload_arxiv = [listing for listing in arxiv_listing if instrument in listing.name]
-            dload_sensfunc = [listing for listing in sensfunc_listing 
-                              if instrument in listing.name]
+        # Access the repo; use a token if one is available
+        if os.getenv('GITHUB_TOKEN') is None:
+            msgs.warn('GITHUB_TOKEN environmental variable is not defined, meaning script will '
+                      'not authenticate a GitHub user via an OAuth token.  Beware of rate limits!')
+            auth = None
+        else:
+            auth = github.Auth.Token(os.getenv('GITHUB_TOKEN'))
+        repo = github.Github(auth=auth).get_repo("pypeit/PypeIt")
 
-            # If not matches, warn and continue
-            if not dload_arxiv:
-                msgs.warn(f"No reid_arxiv files for spectrograph `{instrument}` "
-                          f"found on GitHub in tree '{tag}'.")
+        # Cycle through all the data paths that use github as their host and
+        # collect the contents of the directory
+        msgs.info('Searching github repository ... ')
+        contents = {}
+        for datadir, meta in github_paths.items():
+            if datadir not in selected_paths:
                 continue
+            # Recursively get the directory contents
+            contents[meta['path']] \
+                = cache.github_contents(repo, branch, f'pypeit/data/{meta["path"]}')
+        msgs.info('Searching github repository ... done.')
+        
+        # Determine which files should be in the cache (or in the repo)
+        # TODO: This is currently broken because not all spectrograph-dependent
+        # files are named after their *exact* pypeit spectrograph names.  E.g.,
+        # the reid_arxiv file for gemini_flamingos2 is
+        # ``Flamingos2_HK_HK.fits``, not ``gemini_flamingos2_HK_HK.fits``.  It's
+        # a major effort to rename all those files.  The other (better?) option
+        # would be to put all these reid_arxiv files into subdirectories named
+        # after each spectrograph.
```

Comment:
I suspect both might be better, but I'm a folder enthusiast... In the example you give, I would do:
`gemini_flamingos2/gemini_flamingos2_HK_HK.fits`
but, I agree that this would be tedious...

Reply:
Leaving as a TODO for now.